rclpy/src/rclpy/_rclpy.c

```diff
@@ -370,17 +703,46 @@ rclpy_get_ready_subscriptions(PyObject * Py_UNUSED(self), PyObject * args)
     return NULL;
   }
 
-  size_t sub_idx;
-  PyObject * sub_ready_list = PyList_New(0);
-  for (sub_idx = 0; sub_idx < wait_set->size_of_subscriptions; sub_idx++) {
-    if (wait_set->subscriptions[sub_idx]) {
-      PyList_Append(
-        sub_ready_list,
-        PyLong_FromUnsignedLongLong((uint64_t)&wait_set->subscriptions[sub_idx]->impl));
+  size_t idx;
+  size_t idx_max;
```

Comment:
Yeah the goal of putting everything in the same function was to use the same for loop for every `entity_type` but couldn't find a proper way to cast the pointers on the fly in pure C without having the compiler complaining. Not very satisfied with the result. If we end up having duplicated variable declarations and loops maybe I should go back to separate functions (as it was before https://github.com/ros2/rclpy/pull/51/commits/65fc902a846a59aaa5aa9d5d942c521ef3ed4d37?diff=split) what do you think ?